lib/src/component_declaration/builder_helpers.dart

```diff
@@ -139,42 +133,55 @@ extension UiPropsMeta on UiProps {
   ///
   /// This can be used to derive consumed props by usage in conjunction with [addUnconsumedProps]
   /// and [addUnconsumedDomProps].
-  PropsMetaCollection get staticMeta => $meta;
-
+  @toBeGenerated PropsMetaCollection get staticMeta => throw UngeneratedError(member: #meta);
 
-  /// Copies props from the provided [props] instance to this [UiProps] instance,
-  /// filtering out props found in [consumedProps].
+  /// Copies key-value pairs from the provided [props] map into this map,
+  /// excluding those with keys found in [consumedProps].
   ///
   /// [consumedProps] should be a `Iterable<PropsMeta>` instance.
-  /// This is the return type of [PropsMetaCollection]'s related APIs `forMixins`
-  /// and `allExceptForMixins`. `forMixin` can be used easily by then calling `.inList()`
-  /// on the returned [PropsMeta] instance.
+  /// This is the return type of [PropsMetaCollection]'s related APIs `forMixins`,
+  /// `allExceptForMixins`, and `all`.
+  ///
+  /// __Example:__
+  ///
+  /// ```dart
+  /// // within a functional component (wrapped in `uiFunction`)
+  /// // Consider props in FooProps "consumed"...
+  /// final consumedProps = props.staticMeta.forMixins({FooProps});
+  /// // ...and filter them out when forwarding props to Bar.
+  /// return (Bar()..addUnconsumedProps(props, consumedProps))();
+  /// ```
   ///
   /// To only add DOM props, use [addUnconsumedDomProps].
   ///
   /// Related: [UiComponent2]'s `addUnconsumedProps`
   void addUnconsumedProps(Map props, Iterable<PropsMeta> consumedProps) {
```

Comment:
Is there a a reason these are only implemented in the builder_helpers subclass of UiProps and not the component_base version? Since they don't depend directly on generated code, I think they should be moved to the base class.

If we did that, we also wouldn't have to add new `throw UnimplementedError()` overrides to classes that don't extend from this one.

Reply:
Nope, good call!